test/model/update.test.ts

```diff
@@ -510,3 +510,72 @@ test('exposes a root entity for a derivitive value of a nested property', () =>
 
   expect(result).toHaveProperty(['address', 'billing', 'country'], 'US')
 })
+
+test('sets properties to and from null', () => {
+  const db = factory({
+    user: {
+      id: primaryKey(datatype.uuid),
+      firstName: nullable(name.firstName),
+    },
+  })
+
+  db.user.create({
+    id: 'abc-123',
+    firstName: null,
+  })
+
+  expect(
+    db.user.update({
+      where: {
+        id: {
+          equals: 'abc-123',
+        },
+      },
+      data: {
+        firstName: 'John',
+      },
+    }),
+  ).toHaveProperty('firstName', 'John')
+
+  expect(
+    db.user.update({
+      where: {
+        id: {
+          equals: 'abc-123',
+        },
+      },
+      data: {
+        firstName: null,
+      },
+    }),
+  ).toHaveProperty('firstName', null)
+})
+
+test('throws when setting a non-nullable property to null', () => {
+  const db = factory({
+    user: {
+      id: primaryKey(datatype.uuid),
+      firstName: name.firstName,
+    },
+  })
+
+  db.user.create({
+    id: 'abc-123',
+  })
+
+  expect(() =>
+    db.user.update({
+      where: {
+        id: {
+          equals: 'abc-123',
+        },
+      },
+      data: {
+        // @ts-ignore
```

Comment:
I'd generally prefer `@ts-expect-error` with a short description (like you did in one of the files above).